src/main/java/retriever/Parser.java

```diff
@@ -0,0 +1,84 @@
+package retriever;
+
+import retriever.exception.IllegalCommandException;
+import retriever.exception.RetrieverException;
+import retriever.task.TaskList;
+
+/**
+ * The Parser class is responsible for parsing
+ * various user inputs.
+ */
+public class Parser {
+    private TaskList taskList;
+    private boolean isSessionDone;
+
+    /**
+     * Initializes the task list and marks the beginning of a session.
+     *
+     * @param taskList The Object through which various task operations can be performed.
+     */
+    public Parser(TaskList taskList) {
+        this.taskList = taskList;
+        this.isSessionDone = false;
+    }
+
+    /**
+     * Marks the isSessionDone to true, when the user inputs
+     * the "bye" keyword, indicating, the end of session.
+     */
+    public void markSessionDone() {
+        isSessionDone = true;
+    }
+
+    /**
+     * Returns whether the user would like to continue or no.
+     *
+     * @return A boolean indicating to end the session or not.
+     */
+    public boolean isExit() {
+        return isSessionDone;
+    }
```

Comment:
should this two functions' names be similar since they are working on the same variable?